### PR TITLE
Disabling call butons after call for 1.5sec

### LINKED ISSUE #850: Little enhancement - Disabling call button for a second or two
Hi just wanted to pass on a small usability improvement.  When someone hangs up a call, it may be good to disable the call/hangup button for a second or two for all participants.  Just had an example where one party hung up, but the other party hung up as well, while the event i guess was going across the wire.  The call hung up, but then got a call back straight away to start the call again..

Thanks

### PR DESCRIPTION
**Pull request by agilob**
closes #850

### CODE REVIEW

File: src/widget/form/chatform.cpp [hunk 1]
```diff
@@ -648,10 +559,9 @@ void ChatForm::onRejectCallTriggered()
...
     audioOutputFlag = false;
     emit rejectCall(callId);
-    micButton->setObjectName("green");
-    micButton->style()->polish(micButton);
-    volButton->setObjectName("green");
-    volButton->style()->polish(volButton);
+    
+    enableCallButtons();
+
 }
```
Comment (tux3) [inline src/widget/form/chatform.cpp:567]: There's a enableCallButtons here, but not in onHangupCallTriggered just above, is this intended?

File: src/widget/form/chatform.cpp [hunk 2]
```diff
@@ -686,37 +596,77 @@ void ChatForm::onAvCallFailed(int FriendId)
...
+    connect(videoButton, SIGNAL(clicked()),
+            this, SLOT(onVideoCallTriggered()));
+    connect(micButton, SIGNAL(clicked()),
+            this, SLOT(onMicMuteToggle()));
+    connect(volButton, SIGNAL(clicked()),
+            this, SLOT(onVolMuteToggle()));
+    
+    disableCallButtonsTimer->stop();
+    disableCallButtonsTimer = nullptr;
+    delete disableCallButtonsTimer;
```
Comment (tux3) [inline src/widget/form/chatform.cpp:670]: You never actually delete it, you're deleting nullptr here :)

File: src/widget/form/chatform.cpp [hunk 3]
```diff
@@ -910,20 +860,21 @@ void ChatForm::startCounter()
 
 void ChatForm::stopCounter()
 {
-    if (timer)
-    {
+    if (callDurationTimer)
+    {
```
Comment (tux3) [inline src/widget/form/chatform.cpp]: Just nitpicking, but there are trailing spaces here.

File: src/widget/form/chatform.cpp [hunk 4]
```diff
@@ -910,20 +860,21 @@ void ChatForm::startCounter()
...
                              "white", QDateTime::currentDateTime());
-        timer->stop();
+        callDurationTimer->stop();
         callDuration->setText("");
         callDuration->hide();
-        timer = nullptr;
-        delete timer;
+        callDurationTimer = nullptr;
+        
+        delete callDurationTimer;
```
Comment (tux3) [inline src/widget/form/chatform.cpp:874]: And you're deleting nullptr again :)

**Comment by tux3** [comment]
I still want to test it extensively before merging since this part of the code is a little ugly and prone to bugs, but first there are some simple things to fix.

**Comment by agilob** [comment]
Sure, this issue waited 2months, so it can wait a few days more ;)

**Comment by tux3** [comment]
You forgot to add volButtonDisabled.png to the ressources, so it would just disappear at runtime, otherwise it's all good. I'll merge this later today.

**Pull request merged**